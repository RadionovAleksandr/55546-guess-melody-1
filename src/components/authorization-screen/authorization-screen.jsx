import React from "react";

import {connect} from "react-redux";
import propTypes from "prop-types";

import {Operation} from "../../reducer/user/user";

class AuthorizationScreen extends React.PureComponent {
  constructor(props) {
    super(props);
    this._loginField = React.createRef();
    this._passwordField = React.createRef();
    this._errorLogin = React.createRef();
    this._errorPassword = React.createRef();

    this.checkDataSignIn = this.checkDataSignIn.bind(this);
  }

  checkDataSignIn(email, password, errorBlocks) {
    if (!email) {
      errorBlocks.email.innerHTML = `Не корректный логин!`;
      errorBlocks.login.style.display = `block`;
    } else if (!password) {
      errorBlocks.password.innerHTML = `Не корректный пароль!`;
      errorBlocks.password.style.display = `block`;
    } else {
      this.props.signIn({email, password});
    }
  }

  render() {
    const {
      _loginField,
      _passwordField,
      _errorLogin,
      _errorPassword,
      checkDataSignIn
    } = this;

    return <section className="login" style={{zIndex: 999, marginTop: `-200px`}}>
      <div className="login__logo">
        <img src="img/melody-logo.png" alt="Угадай мелодию" width="186" height="83" />
      </div>
      <h2 className="login__title">Необходима авторизация</h2>
      <p className="login__text">Представтесь!</p>
      <form className="login__form" action="">
        <p className="login__field">
          <label className="login__label" htmlFor="name">Логин</label>
          <input ref={this._loginField} className="login__input" type="text" name="name" id="name" />
          <span ref={this._errorLogin} className="login__error">Неверный логин</span>
        </p>
        <p className="login__field">
          <label className="login__label" htmlFor="password">Пароль</label>
          <input ref={this._passwordField} className="login__input" type="text" name="password" id="password" />
          <span ref={this._errorPassword} className="login__error">Неверный пароль</span>
        </p>
        <button className="login__button button" type="submit" onClick={(evt) => {
          evt.preventDefault();
          if (_loginField && _passwordField) {
            checkDataSignIn(
                _loginField.current.value,
                _passwordField.current.value, {
                  login: _errorLogin.current,
                  password: _errorPassword.current,
                });
          }
        }}>Войти</button>
      </form>
    </section>;
  }
}

AuthorizationScreen.propTypes = {
  signIn: propTypes.func.isRequired
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps);

const mapDispatchToProps = (dispatch) => ({
  signIn: (data) => {
    dispatch(Operation.signIn(data));
  },
});

export {AuthorizationScreen};

export default connect(mapStateToProps, mapDispatchToProps)(AuthorizationScreen);
