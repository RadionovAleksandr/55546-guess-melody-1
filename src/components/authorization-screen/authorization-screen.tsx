import * as React from "react";

import {connect} from "react-redux";

import {Operation} from "../../reducer/user/user";

interface DataSignIn {
  email: string,
  password: string
}

interface Props {
  signIn: (data: DataSignIn) => void
}

class AuthorizationScreen extends React.PureComponent<Props, null> {
  private _loginField: React.RefObject<HTMLInputElement>;
  private _passwordField: React.RefObject<HTMLInputElement>;
  private _errorLogin: React.RefObject<HTMLDivElement>;
  private _errorPassword: React.RefObject<HTMLDivElement>;

  constructor(props: Props) {
    super(props);
    this._loginField = React.createRef();
    this._passwordField = React.createRef();
    this._errorLogin = React.createRef();
    this._errorPassword = React.createRef();

    this.checkDataSignIn = this.checkDataSignIn.bind(this);
  }

  checkDataSignIn(email: string, password: string, errorBlocks: any) {
    if (!email) {
      errorBlocks.email.innerHTML = `Не корректный логин!`;
      errorBlocks.login.style.display = `block`;
    } else if (!password) {
      errorBlocks.password.innerHTML = `Не корректный пароль!`;
      errorBlocks.password.style.display = `block`;
    } else {
      this.props.signIn({email, password});
    }
  }

  render() {
    const {
      _loginField,
      _passwordField,
      _errorLogin,
      _errorPassword,
      checkDataSignIn
    } = this;

    return <section className="login">
      <div className="login__logo">
        <img src="img/melody-logo.png" alt="Угадай мелодию" width="186" height="83" />
      </div>
      <h2 className="login__title">Необходима авторизация</h2>
      <p className="login__text">Представтесь!</p>
      <form className="login__form" action="">
        <p className="login__field">
          <label className="login__label" htmlFor="name">Логин</label>
          <input ref={this._loginField} defaultValue="test@test.com" className="login__input" type="text" name="name" id="name" />
          <span ref={this._errorLogin} className="login__error">Неверный логин</span>
        </p>
        <p className="login__field">
          <label className="login__label" htmlFor="password">Пароль</label>
          <input ref={this._passwordField} className="login__input" type="text" name="password" id="password" />
          <span ref={this._errorPassword} className="login__error">Неверный пароль</span>
        </p>
        <button className="login__button button" type="submit" onClick={(evt) => {
          evt.preventDefault();
          if (_loginField && _passwordField) {
            checkDataSignIn(
                _loginField.current.value,
                _passwordField.current.value, {
                  login: _errorLogin.current,
                  password: _errorPassword.current,
                });
          }
        }}>Войти</button>
      </form>
    </section>;
  }
}

const mapStateToProps = (state: any, ownProps: any) => Object.assign({}, ownProps);

const mapDispatchToProps = (dispatch) => ({
  signIn: (data: DataSignIn) => {
    dispatch(Operation.signIn(data));
  },
});

export {AuthorizationScreen};

export default connect(mapStateToProps, mapDispatchToProps)(AuthorizationScreen);
