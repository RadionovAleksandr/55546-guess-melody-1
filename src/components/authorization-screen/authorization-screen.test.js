import React from 'react';
import renderer from 'react-test-renderer';

import {AuthorizationScreen} from './authorization-screen';

describe(`Test authorizationScreen`, () => {
  it(`renderer`, () => {
    const tree = renderer.create(<AuthorizationScreen
      signIn={jest.fn()}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
