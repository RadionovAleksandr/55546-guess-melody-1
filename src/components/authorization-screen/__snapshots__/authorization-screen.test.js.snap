// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test authorizationScreen renderer 1`] = `
<section
  className="login"
  style={
    Object {
      "marginTop": "-200px",
      "zIndex": 999,
    }
  }
>
  <div
    className="login__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="login__title"
  >
    Необходима авторизация
  </h2>
  <p
    className="login__text"
  >
    Представтесь!
  </p>
  <form
    action=""
    className="login__form"
  >
    <p
      className="login__field"
    >
      <label
        className="login__label"
        htmlFor="name"
      >
        Логин
      </label>
      <input
        className="login__input"
        id="name"
        name="name"
        type="text"
      />
      <span
        className="login__error"
      >
        Неверный логин
      </span>
    </p>
    <p
      className="login__field"
    >
      <label
        className="login__label"
        htmlFor="password"
      >
        Пароль
      </label>
      <input
        className="login__input"
        id="password"
        name="password"
        type="text"
      />
      <span
        className="login__error"
      >
        Неверный пароль
      </span>
    </p>
    <button
      className="login__button button"
      onClick={[Function]}
      type="submit"
    >
      Войти
    </button>
  </form>
</section>
`;
